File: benchmarks/fib_rec/fib_rec.cpp
#include <cstdio>
#include <cstdint>

static int64_t fib(int64_t n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main() {
    const int64_t TARGET = 30, REPEAT = 20;
    int64_t result = 0;
    for (int64_t i = 0; i < REPEAT; i++) result = fib(TARGET);
    std::printf("%lld\n", (long long)result);
}

File: benchmarks/fib_rec/fib_rec.rs
fn fib(n: i64) -> i64 {
    if n < 2 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    const TARGET: i64 = 30;
    const REPEAT: i64 = 20;
    let mut result = 0;
    for _ in 0..REPEAT {
        result = fib(TARGET);
    }
    println!("{}", result);
}
